modules/ROOT/pages/tuning-test-validations.adoc

```diff
@@ -0,0 +1,85 @@
+
+.Performance test infrastructure (Separate host avoid interference)
+image::mruntime-tuning-test-infrastructure.png[Performance test infrastructure]
+
+=== Choose the Right Tools
+
+The components used for testing can make a big difference for your use case:
+
+* To get real performance data, choose the most suitable load injector and a backend service configured with the same setup as your customer environment.  +
+MuleSoft Performance team uses Apache JMeter; however, there are many enterprise and open source load test tools in the market with different implementations such as single-threaded, multi-threaded, non-blocking, etc. Each one has its advantages and disadvantages depending on the scenario.
+* After you configure the tools, continue to use them for consistent comparisons.
+* If you change any of the tools, execute all the tests again, and create a new baseline.
+
+=== Isolate the Components of the Test
+Because performance is an iterative process and many variables affects the results, it is important that you try to isolate the problem and control these variables. +
+To improve the results, split the application into several small ones targeting a specific component or connector. This helps to spot the bottlenecks and direct optimization efforts towards a specific flow element.
+
+
+=== Use Representative Workloads
+The combination of the following practices may affect the performance of your Mule app. +
+When you test your application, it is important to imitate real world customer use cases. To obtain customer scenarios, analyze:
+
+* User behavior
+* Workload (virtual users, messages, etc)
+* Payloads (type, size, complexity)
+
+Additionally, as a best practice:
+
+* Create different scenarios with a repeatable, common, and high load in order to stress Mule, but not too high that pushes Mule to the limit and causes test failure.
+
+* To imitate actual customer behavior, introduce variables as think time between requests and latency to the backend service.
+
+
+
+
+== Execute Performance Test Validations
+
+Before you start the test execution, ideally remove the ramp-up, tear down, and error percentage settings that can affect the throughput calculation results. Considering the best practices from the previous section, execute the testing:
+
+. Apply performance testing to spot and validate the bottlenecks of the applications.
+. Use an iterative approach for tuning issues that you find:
+.. First, tune Mule’s flows and configurations (only for Cloudhub and on-premises applications)
+.. Tune the JVM, GC, and so on. (only for on-premises applications)
+.. Finally, tune operating system variables (only for On-premises applications)
+. Execute the test several times for trustworthy results. Make one xref:tuning-recommendations.adoc[Tuning Recommendation] change at a time and then retest.
+. Check the test results.
```

Comment:
At this point, I am not sure that these testing steps are in order:
. Test to find issues
. Tune issues
. Retest to find...issues not fixed by the tuning?
. Check the test results

Do we really mean:
. Test to find issues.
. For the first issue, change the value of the first item in the Mule configuration. 
. Retest and check the results.
. Continue changing values one at a time and retesting until there are no issues in the Mule configuration.
. Repeat Steps 2 through 4 in on-premises applications for:
* JVC, GC, and so on
* Operating system variables
. Move to the next issue and repeat Steps 2 through 5, until no issues remain.

So, there is iteration of tuning and testing within each issue, and iteration of this process for each issue, until the test is successful?

Whew! I would like to see a graphic illustrating this.

Please adjust this subtopic as you see fit, but please also run it by a peer writer to see if it makes sense to them. I feel a little out of my depth here.